bin/src/hickory-dns.rs

```diff
@@ -895,3 +896,41 @@ fn logger(level: tracing::Level) -> Result<(), String> {
 
     Ok(())
 }
+
+
+/// Build a TcpListener for a given IP, port pair; IPv6 listeners will not accept v4 connections
+fn build_tcp_listener(ip: &IpAddr, port: u16) -> Result<tokio::net::TcpListener, std::io::Error> {
```

Comment:
IMO we should not have commits that add unused private API (that is, this would trigger warnings in CI).

I think we should use the unqualified `TcpListener` (already imported). Would also suggest importing `io` and using it for these return types.

Would suggest taking `IpAddr` by value, since it's `Copy` (and not super large), and we're going to dereference it anyway.

Reply:
Can you provide a little more clarity on when and why you want to see functions as standalone commits? I've gotten feedback from you in the past that new functions, or renamed functions, should be in separate commits. At least one of those new functions (the add config group function) was unused at the time, as it was new.

I struggle to see see how this situation is any different.

Reply:
Sure, sorry if I'm not being clear. There's more detail in the rustls [`CONTRIBUTING.md`](https://github.com/rustls/rustls/blob/main/CONTRIBUTING.md) which I wrote large parts of.

My goal is to ease reviewing by having self-consistent commits (that is, each commit should be able to pass CI as far as you know, even though actually testing that in practice would take way too much time) that are easy to review. Easy to review means that (a) each commit does one thing, (b) all the things belonging with the one thing are in one commit. Since I'm reviewing commit-by-commit, if you stick function additions in a different commit from sites using it you end up with (a) warnings for the first commit that we want to avoid, (b) harder to review function definition together with usage sites.

Hope that helps? It's also possible that I let it slip earlier because (a) I might not have been paying attention or (b) I'm trying to not quite overwhelm you with all the feedback at once. (I know it's a lot as is -- sorry -- also feel free to push back if you disagree with the idea/rationale.)

Reply:
That is helpful, thanks.  I think these should be resolved now, but let me know if you disagree.